service.go

```diff
@@ -187,27 +187,25 @@ func (s *Service) createHTTPComponent() (Component, error) {
 	port = strconv.FormatInt(portVal, 10)
 	log.Infof("creating default HTTP component at port %s", port)
 
-	options := []http.OptionFunc{
-		http.Port(int(portVal)),
-	}
+	b := http.NewBuilder().WithPort(int(portVal))
```

Comment:
previously i mentioned 2 changes. the second one is: port is already an int...
```suggestion
	b := http.NewBuilder().WithPort(port)
```

Reply:
`port` is a string as [`strconv.FormatInt`](https://golang.org/pkg/strconv/#FormatInt) returns a string. Do you want me to add a step that converts to int beforehand? I think it's concise as is.

Reply:
you are correct, I totally confused it with parsing a string to an int...